File: 1.1/src/cz/mg/vulkan/PFNvkDebugReportCallbackEXT.java
package cz.mg.vulkan;

public class PFNvkDebugReportCallbackEXT extends VkFunctionPointer {
    public PFNvkDebugReportCallbackEXT() {
    }

    protected PFNvkDebugReportCallbackEXT(VkMemory vkmemory) {
        super(vkmemory);
    }

    protected PFNvkDebugReportCallbackEXT(VkMemory vkmemory, long vkaddress) {
        super(vkmemory, vkaddress);
    }



    public PFNvkDebugReportCallbackEXT(long value) {
        setValue(value);
    }

    public PFNvkDebugReportCallbackEXT(VkInstance instance) {
        super(instance, new VkString("vkDebugReportCallbackEXT"));
    }

    public int call(int flags, int objectType, long object, long location, int messageCode, VkChar pLayerPrefix, VkChar pMessage, VkObject pUserData){
        return callNative(getValue(), flags, objectType, object, location, messageCode, pLayerPrefix != null ? pLayerPrefix.getVkAddress() : VkPointer.NULL, pMessage != null ? pMessage.getVkAddress() : VkPointer.NULL, pUserData != null ? pUserData.getVkAddress() : VkPointer.NULL);
    }

    protected static native int callNative(long vkaddress, int flags, int objectType, long object, long location, int messageCode, long pLayerPrefix, long pMessage, long pUserData);

}

File: 1.1/src/c/cz/mg/vulkan/PFN_vkDebugReportCallbackEXT.c
#include <jni.h>
#include <stdlib.h>
#include <string.h>
#include <vulkan/vulkan.h>

jlong jniPointerToLong(const void* p);
void* jniLongToPointer(jlong l);
jlong jniFunctionPointerToLong(PFN_vkVoidFunction p);
PFN_vkVoidFunction jniLongToFunctionPointer(jlong l);
void jniThrowException(JNIEnv* env, const char* message);

jint Java_cz_mg_vulkan_PFNvkDebugReportCallbackEXT_callNative(JNIEnv* env, jclass jc, jlong address, jint flags, jint objectType, jlong object, jlong location, jint messageCode, jlong pLayerPrefix, jlong pMessage, jlong pUserData){
    (void)env;
    (void)jc;
    PFN_vkDebugReportCallbackEXT f = (PFN_vkDebugReportCallbackEXT)jniLongToFunctionPointer(address);
    return f(
        (VkDebugReportFlagsEXT)flags,
        (VkDebugReportObjectTypeEXT)objectType,
        (uint64_t)object,
        (size_t)location,
        (int32_t)messageCode,
        ((char*)jniLongToPointer(pLayerPrefix)),
        ((char*)jniLongToPointer(pMessage)),
        ((void*)jniLongToPointer(pUserData))
    );
}

